akka-stream/src/main/scala/akka/stream/javadsl/SubSource.scala

```diff
@@ -1490,6 +1490,30 @@ class SubSource[Out, Mat](
   def concat[M](that: Graph[SourceShape[Out], M]): SubSource[Out, Mat] =
     new SubSource(delegate.concat(that))
 
+  /**
+   * Concatenate the given [[Source]]s to this [[Flow]], meaning that once this
+   * Flow’s input is exhausted and all result elements have been generated,
+   * the Source’s elements will be produced.
+   *
+   * Note that the [[Source]]s are materialized together with this Flow and is "detached" meaning it will
+   * in effect behave as a one element buffer in front of both the sources, that eagerly demands an element on start
+   * (so it can not be combined with `Source.lazy` to defer materialization of `that`).
+   *
+   * The second source is then kept from producing elements by asserting back-pressure until its time comes.
+   *
+   * When needing a concat operator that is not detached use [[#concatLazyMat]]
+   *
+   * '''Emits when''' element is available from current stream or from the given [[Source]] when current is completed
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' given [[Source]] completes
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def concatAll(those: Graph[SourceShape[Out], _]*): SubSource[Out, Mat] =
```

Comment:
To be compatible from Java there must be a `varargs` annotation:
```
@varargs
def concatAll(...
```

Reply:
and `@SafeVarargs` .

I am currently not sure should this `all` variances go with `varargs` or not. When it especially when there are more than one parameters.

If we go with the `varargs` ,then the parameter must come to be the last. and then the order them are not alias the the not-all one, otherwise , we need either `immutable.Seq` or make it curried. I first rise this question in https://github.com/akka/akka/pull/31298 , so I make all the others PR as draft. I want to alias the same style.

Reply:
If the style is settled, then I will first go and finish the https://github.com/akka/akka/pull/31292

Reply:
Btw ,the contributor guide requires alias the `java.util.List` with `immutable.Seq`.

Reply:
In Reactor-core :
```java
public static <T> Flux<T> concat(Iterable<? extends Publisher<? extends T>> sources)
```
https://github.com/reactor/reactor-core/blob/main/reactor-core/src/main/java/reactor/core/publisher/Flux.java#L418

Reply:
I think it should be varargs for methods where there are no other additional parameters and fall back to list when there is a need to have additional other parameters, unless the additional parameters are important enough to go first, but I don't think that is the case for `eagerCancel` for example.

Reply:
Thanks for your input, I'm on vacation,will update PR asap.